sampler.go

```diff
@@ -222,23 +237,50 @@ func (s *guaranteedThroughputProbabilisticSampler) IsSampled(id uint64, operatio
 	return sampled, s.tags
 }
 
-func (s *guaranteedThroughputProbabilisticSampler) Close() {
+// Close implements Close() of Sampler.
+func (s *GuaranteedThroughputProbabilisticSampler) Close() {
 	s.probabilisticSampler.Close()
 	s.lowerBoundSampler.Close()
 }
 
-func (s *guaranteedThroughputProbabilisticSampler) Equal(other Sampler) bool {
-	if o, ok := other.(*guaranteedThroughputProbabilisticSampler); ok {
+// Equal implements Equal() of Sampler.
+func (s *GuaranteedThroughputProbabilisticSampler) Equal(other Sampler) bool {
+	if o, ok := other.(*GuaranteedThroughputProbabilisticSampler); ok {
 		return s.operation == o.operation && s.probabilisticSampler.Equal(o.probabilisticSampler) &&
 			s.lowerBoundSampler.Equal(o.lowerBoundSampler)
 	}
 	return false
 }
 
+func (s *GuaranteedThroughputProbabilisticSampler) update(lowerBound, samplingRate float64) error {
+	if s.samplingRate != samplingRate {
```

Comment:
add comment to the method explaining why this does not use locks even though it mutates the struct